File: Backend/src/main/java/com/autostock/service/ProveedorService.java
package com.autostock.service;

import com.autostock.dto.ProveedorDTO;
import java.util.List;

public interface ProveedorService {

    ProveedorDTO crear(ProveedorDTO proveedorDTO);

    List<ProveedorDTO> listar();

    ProveedorDTO obtenerPorId(Long id);

    ProveedorDTO actualizar(Long id, ProveedorDTO proveedorDTO);

    void eliminar(Long id);
}

File: Frontend/src/services/proveedorService.js
import api from './api';

const BASE = '/api/proveedores';

export const proveedorService = {
  listar: () => api.get(BASE),
  crear: (data) => api.post(BASE, data),
  actualizar: (id, data) => api.put(`${BASE}/${id}`, data),
  eliminar: (id) => api.delete(`${BASE}/${id}`)
};
